pkg/odo/cli/component/create.go

```diff
@@ -859,12 +860,18 @@ func (co *CreateOptions) devfileRun() (err error) {
 		return errors.Wrap(err, "failed to download project for devfile component")
 	}
 
-	// save devfile
+	// save devfile and corresponding resources if possible
 	// use original devfileData to persist original formatting of the devfile file
 	err = ioutil.WriteFile(DevfilePath, devfileData, 0644) // #nosec G306
 	if err != nil {
 		return errors.Wrapf(err, "unable to save devfile to %s", DevfilePath)
 	}
+	if co.devfileMetadata.devfilePath.value == "" && !devfileExist && !strings.Contains(co.devfileMetadata.devfileRegistry.URL, "github") {
+		err = registryLibrary.PullStackFromRegistry(co.devfileMetadata.devfileRegistry.URL, co.devfileMetadata.componentType, co.componentContext)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
@GeekArthur Why do you rewrite this piece of code when it is already well defined here at [L820](https://github.com/openshift/odo/pull/4636/files#diff-e9cbb4b32525048f3bf6ffce25bb7e24c02221819d5b9de9347fe2c835828058R820)?
```go
devfileExist := util.CheckPathExists(DevfilePath)
if co.devfileMetadata.devfilePath.value != "" {
...
} else {
    if devfileExist {
    ...
    } else {
        if strings.Contains(co.devfileMetadata.devfileRegistry.URL, "github") {
        ...
        } else {
            err = registryLibrary.PullStackFromRegistry(co.devfileMetadata.devfileRegistry.URL, co.devfileMetadata.componentType, co.componentContext)
            if err != nil {
		return err
	    }
	}
    }
}
```    
The above logic seems to be the same as the one you have written - 
```go
if co.devfileMetadata.devfilePath.value == "" && !devfileExist && !strings.Contains(co.devfileMetadata.devfileRegistry.URL, "github") {
		err = registryLibrary.PullStackFromRegistry(co.devfileMetadata.devfileRegistry.URL, co.devfileMetadata.componentType, co.componentContext)
		if err != nil {
			return err
		}
	}
```

Do we need to rewrite the devfile? This seems to be a problem because it rewrites the devfile that was already written at [L865](https://github.com/openshift/odo/pull/4636/files#diff-e9cbb4b32525048f3bf6ffce25bb7e24c02221819d5b9de9347fe2c835828058R865).